### PR TITLE
[TechDraw] Improve readability of App scripts, part 5

### PR DESCRIPTION
**Pull request by benj5378**
### [TechDraw] Improve readability of App scripts, part 5

- [[TechDraw] Improve readability of EdgeWalker.cpp](https://github.com/FreeCAD/FreeCAD/commit/1e503fca3518fd21f44f4b9b6f3bc5717c9aa6f4)
- [[TechDraw] Improve readability of GeometryObject.cpp](https://github.com/FreeCAD/FreeCAD/commit/3a7a97d6ada1b5d99703f8f110aec543f695cf08)
- [[TechDraw] Improve readability of GeomFormatPyImp.cpp](https://github.com/FreeCAD/FreeCAD/commit/50ec806fb99c0e23f698addcca68090e9d51637a)
- [[TechDraw] Improve readability of HatchLine.cpp](https://github.com/FreeCAD/FreeCAD/commit/a9b0f7e4a51c1121263bfb5f4e29842389d16e04)

Thank you for creating a pull request to contribute to FreeCAD! To ease integration, we ask you to conform to the following items. Pull requests which don't satisfy all the items below might be rejected. If you are in doubt with any of the items below, don't hesitate to ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10)!

- [x]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [x]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [x]  Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [ ]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [x]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [x]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [x]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the issue ID number from our [Issues database](https://github.com/FreeCAD/FreeCAD/issues) in case a particular commit solves or is related to an existing issue. Ex: `Draft: fix typos - fixes #4805`

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.20 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=56135).

---

### CODE REVIEW

**Comment by berndhahnebach** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7210">pipeline status</a> for feature branch [PR_7210](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7210). Pipeline [589897094](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/589897094) was triggered at [a9b0f7e](https://github.com/FreeCAD/FreeCAD/pull/7210/commits/a9b0f7e4a51c1121263bfb5f4e29842389d16e04). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: src/Mod/TechDraw/App/HatchLine.cpp [hunk 1]
```diff
@@ -186,40 +176,41 @@ Base::Vector3d LineSet::getPatternStartPoint(TechDraw::BaseGeomPtr g, double &of
...
+        }
+    } else if (lenStartOrg > lenEndOrg) {                          //origin is after end
+        double c = ceil(patsEndOrg);
+        if (c <= patsStartOrg) {                      //c is an integer pattern count in [patsStartOrg,patsEndOrg]
+            offset = 0.0;
+            return lineOrigin - c*patternLength*getUnitDir();
         } else {
-            Base::Console().Log("ERROR - HL::getPatternStart - something has gone wrong!\n");
+            offset = ceil(patsStartOrg) - patsStartOrg;      //fraction of a patternLength patstartorg to repeat point
+            offset = offset * patternLength;
```
Comment (benj5378) [inline src/Mod/TechDraw/App/HatchLine.cpp:207]: Won't this overwrite offset?

File: src/Mod/TechDraw/App/HatchLine.cpp [hunk 2]
```diff
@@ -186,40 +176,41 @@ Base::Vector3d LineSet::getPatternStartPoint(TechDraw::BaseGeomPtr g, double &of
...
+            return gStart;
         }
     }
-    return result;
+    
+    Base::Console().Log("ERROR - HL::getPatternStart - something has gone wrong!\n");
+    return getOrigin();
```
Comment (benj5378) [inline src/Mod/TechDraw/App/HatchLine.cpp:213]: Assuming that `getOrigin()` will still return the same value as if it had been called in the start of the function

**Comment by benj5378** [comment]
Ready for review @WandererFan

**Comment by berndhahnebach** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7210">pipeline status</a> for feature branch [PR_7210](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7210). Pipeline [600453088](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/600453088) was triggered at [60e3a16](https://github.com/FreeCAD/FreeCAD/pull/7210/commits/60e3a1683c573de8cbf4f17c8df526df40502a38). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: src/Mod/TechDraw/App/HatchLine.cpp (hunk 1 above)
Comment (WandererFan) [reply src/Mod/TechDraw/App/HatchLine.cpp:207]: It is supposed to overwrite offset.  I guess it should really be returning std::pair<Base::Vector3d, double>.

File: src/Mod/TechDraw/App/HatchLine.cpp (hunk 2 above)
Comment (WandererFan) [reply src/Mod/TechDraw/App/HatchLine.cpp:213]: getOrigin applies to a single record from the PAT file, so it is essentially a constant while we are making the hatch lines that are generated by that record.

**Comment by benj5378** [comment]
@WandererFan Let me know what needs to be done to move forward with this

**Comment by berndhahnebach** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7210">pipeline status</a> for feature branch [PR_7210](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7210). Pipeline [630738991](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/630738991) was triggered at [b930641](https://github.com/FreeCAD/FreeCAD/pull/7210/commits/b9306419936c1c7a2d6d308cac8e1848ef2fa1c3). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by WandererFan** [comment]
closed as stale branch with merge conflicts.

**Pull request closed without merging**